core/src/storage2/lazy/lazy_hmap.rs

```diff
@@ -395,7 +394,7 @@ where
     Key: From<<H as Hasher>::Output>,
 {
     /// Gets the given key's corresponding entry in the map for in-place manipulation.
-    pub fn entry(&mut self, key: K) -> Entry<K, V, H> {
+    pub fn entry(&mut self, key: K) -> Entry<K, V> {
```

Comment:
The `unsafe` scope is bigger than it needs to be and also you are simply loading the entry from the cache while forgetting to eventually load it from storage first.